lib/usd/translators/meshWriter_BlendShapes.cpp

```diff
@@ -0,0 +1,960 @@
+            CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+            for (unsigned int j = 0; j < weightInfo.targetMeshes.length(); ++j) {
+                MObject targetMesh = weightInfo.targetMeshes[j];
+                stat = fnNode.setObject(targetMesh);
+                CHECK_MSTATUS_AND_RETURN_IT(stat);
+                MPlug plgWorldMeshes = fnNode.findPlug("worldMesh", false, &stat);
+                CHECK_MSTATUS_AND_RETURN_IT(stat);
+                unsigned int numWorldMeshesConnected = plgWorldMeshes.numElements();
+                if (numWorldMeshesConnected != 1) {
+                    return MStatus::kFailure;
+                }
+                MPlug plgWorldMesh = plgWorldMeshes.elementByPhysicalIndex(0, &stat);
+                CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+                if (!plgWorldMesh.isSource()) {
+                    return MStatus::kFailure;
+                }
+                MPlugArray destPlugs;
+                plgWorldMesh.destinations(destPlugs);
+                for (unsigned int k = 0; k < destPlugs.length(); ++k) {
+                    MPlug   destPlug = destPlugs[k];
+                    MObject blendShape = destPlug.node();
+                    MPlug   inputTargetGroupPlug;
+                    MPlug   inputTargetPlug;
+                    if (blendShape == curBlendShape) {
+                        MPlug inputTargetItemPlug = destPlug.parent();
+                        MPlug inputTargetItemsPlug = inputTargetItemPlug.array();
+                        inputTargetGroupPlug = inputTargetItemsPlug.parent();
+                        weightInfo.inputTargetGroupIndices.append(
+                            inputTargetGroupPlug.logicalIndex());
+
+                        MPlug inputTargetGroupsPlug = inputTargetGroupPlug.array();
+                        inputTargetPlug = inputTargetGroupsPlug.parent();
+                        weightInfo.inputTargetIndices.append(inputTargetPlug.logicalIndex());
+                    }
+                }
+            }
+
+            // NOTE: (yliangsiew) If the target mesh has "in-between" weights, in
+            // Maya they are stored as an array of sparse ints, where the formula is:
+            // index = fullWeight * 1000 + 5000.
+            // Thus fullWeight values of 0.5, 1.0 and 2.0, they will be connected to
+            // inputTargetItem array indices 5500, 6000 and 7000, respectively.
+            // Refer to the docs for MFnBlendShape::targetItemIndexList for more info.
+            stat = fnBlendShape.targetItemIndexList(
+                weightInfo.weightIndex, deformedMesh, weightInfo.targetItemIndices);
+            CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+            info.weightDatas.push_back(weightInfo);
+        }
+        outInfos.push_back(info);
+    }
+    return stat;
+}
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+VtIntArray getUnionOfVtIntArrays(const VtIntArray* const arrays, const size_t count)
```

Comment:
Am I missing where this is used, or is it not used anywhere? I was wondering whether it could take a `std::vector` or `VtArray` of `VtIntArray` and avoid the separate count, but even better if we can just get rid of it.

Reply:
You're right, this is leftover from a previous implementation attempt I did. Will remove.